integration.py

```diff
@@ -0,0 +1,256 @@
+                            "maximum": 100
+                        },
+                        "total": {
+                            "type": "integer",
+                            "minimum": 0
+                        },
+                        "time": {
+                            "type": "number",
+                            "minumum": 0
+                        }
+                    },
+                    "required": ["error_rate", "total", "time"]
+                }
+            }
+        }
+    },
+    "required": ["results"]
+}
+
+
+class Reporter(object):
+    def __init__(self, is_external=None):
+        self.is_external = is_external or os.environ.get("EXTERNAL")
+        self.client = self._get_client()
+
+    def _get_client(self):
+        if self.is_external:
+            host = os.environ.get("HOST", "localhost")
+            port = os.environ.get("PORT", 5000)
+
+            class Client:
+                @staticmethod
+                def post(route, data, content_type):
+                    headers = {'content-type': content_type}
+                    uri = "http://{}:{}{}".format(host, port, route)
+                    return requests.post(uri, data=data, headers=headers)
+
+            client = Client()
+        else:
+            importlib.reload(app)
+            app.app.config['TESTING'] = True
+            client = app.app.test_client()
+        return client
+
+    def get_text(self, resp):
+        # the response is via `requests`
+        if self.is_external:
+            return resp.text
+        # the response is via `flask.test_client`
+        else:
+            return resp.data.decode('utf-8')
+
+    def validate_sample(self, namespace, name, messages):
+        start = time.time()
+        submission, doc_type, doc_version = (
+            name.split('.batch.json')[0].split('.')
+        )
+        errors = {}
+        for msg in messages:
+            route = '/submit/{}/{}'.format(namespace, doc_type)
+            if int(doc_version) > 0:
+                route = '{}/{}'.format(route, doc_version)
+            resp = self.client.post(route,
+                                    data=msg.encode('utf-8'),
+                                    content_type='application/json')
+            is_error = resp.status_code != 200
+            if is_error:
+                text = self.get_text(resp)
+                errors[text] = errors.get(text, 0) + 1
+
+        end = time.time()
+        error_count = sum(errors.values())
+        total = len(messages)
+        error_rate = error_count / float(total) * 100
+        result = {
+            "{}.{}.{}".format(namespace, doc_type, doc_version): {
+                'error_count': error_count,
+                'total': total,
+                'error_rate': round(error_rate, 2),
+                'time': round(end - start, 2),
+                'errors': errors or None
+            }
+        }
+        return result
+
+    @staticmethod
+    def print(result):
```

Comment:
I didn't know this was even possible, but I dislike it.

Reply:
This uses the Reporter class as a namespace, they seem to logically fit together.

Reply:
I'm basically using the Reporter as a namespace, without needing any of the instance or class variable.

Reply:
I just meant the usage of `def print`, where in python 2 `print` was a keyword and this definitely wasn't possible.